.github/workflows/cypress.yml

```diff
@@ -74,13 +74,11 @@ jobs:
           STEP: ${{ matrix.ci_node_index }}
           CI: true
 
-      - name: Publish test results
-        if: ${{ always() }}
-        uses: mikepenz/action-junit-report@v2.4.2
+      - name: Archive test results
+        uses: actions/upload-artifact@v2
         with:
-          check_name: 'Cypress Tests Summary'
-          github_token: ${{ secrets.GITHUB_TOKEN }}
-          report_paths: 'test-results/e2e-test-output-*.xml'
+          name: cypress-test-results
+          path: test-results
```

Comment:
I had wondered if we could do this with the Mochawesome reports so they could be downloaded all together.

There were some concerns about the [files getting corrupted](https://github.com/actions/upload-artifact#uploading-to-the-same-artifact):
> Warning: Be careful when uploading to the same artifact via multiple jobs as artifacts may become corrupted

But if you haven't experienced any issues, I think it would be cleaner to do the same thing with the Mochawesome test results instead of uploading them under individual names. The reason I think they won't be corrupted is that they have different file names under one folder.

Reply:
Not necessary to make this change in this PR. I could test this myself when I consolidate the changes into one workflow. Just putting this out there in case anyone sees any issues with this.

Reply:
We're also doing this with the videos and screenshots, so as long as those seem fine, I think it makes sense to do the same thing for the Mochawesome results.

Reply:
I may be misunderstanding so please correct me if I am.  Looking at the default artifacts that are generated (the ones that the Mochawesome report is using), they're of `json` format. The ones that I'm uploading here, are generated in a separate directory specifically for the test summary action and are `xml` as that's what the summary action is expecting.  These are serving different purposes, I believe, and can verify it works off the most recent run:

![image](https://user-images.githubusercontent.com/34250637/124186109-d0991480-da89-11eb-8312-71eefc9747ad.png)

(the failures are unrelated)

Reply:
They are different types of test results. I'm just referring to how the Mochawesome reports are uploaded as artifacts like this:

```yml
      - name: Archive Cypress Mochawesome test results
        uses: actions/upload-artifact@v2
        if: ${{ always() }}
        with:
          name: cypress-mochawesome-test-results-artifact-${{ matrix.ci_node_index }}
path: cypress/results
```

But the name could just be `cypress-mochawesome-test-results`. And then instead of downloading 6 or however many reports separately, they could all be downloaded as one folder in the `mochawesome-reports` job. Pretty much doing the same thing you're doing with the XML reports now.

Reply:
Ahh I gotcha.  I'll run that by Holden and see if there was a reason that it was done with the separate artifacts.  If so, we'll put it on deck for a PR after you're done with your workflow consolidation efforts.

Reply:
The reason was that concern about corrupted artifacts I mentioned above, but it seems like that wouldn't be an issue based on how the XML test results are working. I could test it out while I consolidate workflows too.